pkg/aws/policy_document.go

```diff
@@ -33,7 +33,7 @@ type PolicyStatement struct {
 	// federated user to which you would like to allow or deny access. If you are creating an
 	// IAM permissions policy to attach to a user or role, you cannot include this element.
 	// The principal is implied as that user or role.
-	Principal PolicyStatementPrincipal `json:"Principal,omitempty"`
+	Principal *PolicyStatementPrincipal `json:"Principal,omitempty"`
```

Comment:
This is needed so we don't include empty principals when serializing the document